File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.java
class Solution {
    public int[] sortJumbled(int[] mapping, int[] nums) {
        
        List<Integer>ls=new ArrayList<>();
        int n=nums.length;
        for(int i=0;i<n;i++){
            int num=nums[i];
            String num_str=Integer.toString(num);
            int el=0;
            while(num_str.length()>0){
               char ch =num_str.charAt(0);
              num_str= num_str.substring(1);
               el=el*10+mapping[ch-'0'];
            }
            ls.add(el);
        }

        Map<Integer,List<Integer>>mp=new TreeMap<>();

        for(int i=0;i<n;i++){
            int nm=ls.get(i);
            int mm=nums[i];
            if(!mp.containsKey(nm)){
                mp.put(nm,new ArrayList<>());
            }
            mp.get(nm).add(mm);
        }

        List<Integer>ans=new ArrayList<Integer>();
        
        for(Map.Entry<Integer,List<Integer>>it:mp.entrySet()){

            List<Integer>lss=it.getValue();
            while(!lss.isEmpty()){
                ans.add(lss.remove(0));
            }
        }

        int[] result = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++) {
            result[i] = ans.get(i);
        }

        return result;
        



    }
}

File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.cpp
class Solution {
public:
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        
        vector<int>mapped,ans;
        int n=nums.size();
        for(int i=0;i<n;i++){
            int num=nums[i];
            string str_num=to_string(num);

            int el=0;
            while(str_num.size()>0){
                char ch=str_num.front();
                str_num.erase(0,1);
                int con=mapping[ch-'0'];

                el=el*10+con;
            }
            mapped.push_back(el);
        }

        map<int,vector<int>>mp;
        for(int i=0;i<n;i++){
             mp[mapped[i]].push_back(nums[i]);
            
        }
        for(auto x:mp){
            while(x.second.size()>0){
                ans.push_back(x.second[0]);
                x.second.erase(x.second.begin());
            }
        }
        return ans;
    }
};